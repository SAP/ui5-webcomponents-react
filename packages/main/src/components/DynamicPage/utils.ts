export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
