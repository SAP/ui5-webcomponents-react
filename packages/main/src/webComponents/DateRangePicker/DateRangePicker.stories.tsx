import '@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Buddhist.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Islamic.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Japanese.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Persian.js';
import type { Meta, StoryObj } from '@storybook/react';
import CalendarType from '@ui5/webcomponents-base/dist/types/CalendarType.js';
import ValueState from '@ui5/webcomponents-base/dist/types/ValueState.js';
import { DateRangePicker } from './index.js';

const meta = {
  title: 'Inputs / DateRangePicker',
  component: DateRangePicker,
  argTypes: {
    valueStateMessage: { control: { disable: true } }
  },
  args: {
    primaryCalendarType: CalendarType.Gregorian,
    valueState: ValueState.None
  },
  tags: ['package:@ui5/webcomponents']
} satisfies Meta<typeof DateRangePicker>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
