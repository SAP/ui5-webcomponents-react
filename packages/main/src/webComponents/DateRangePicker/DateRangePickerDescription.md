## Usage

The user can enter a date by: Using the calendar that opens in a popup or typing it in directly in the input field (not available for mobile devices). For the `DateRangePicker`