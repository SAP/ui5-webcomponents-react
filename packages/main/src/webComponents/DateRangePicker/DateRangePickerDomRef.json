[
  {
    "name": "dateValue",
    "readonly": "true",
    "visibility": "public",
    "type": "Date",
    "description": "<b>Note:</b> The getter method is inherited and not supported. If called it will return an empty value."
  },
  {
    "name": "dateValueUTC",
    "readonly": "true",
    "visibility": "public",
    "type": "Date",
    "description": "<b>Note:</b> The getter method is inherited and not supported. If called it will return an empty value."
  },
  {
    "name": "endDateValue",
    "readonly": "true",
    "visibility": "public",
    "type": "Date",
    "description": "Returns the end date of the currently selected range as JavaScript Date instance."
  },
  {
    "name": "startDateValue",
    "readonly": "true",
    "visibility": "public",
    "type": "Date",
    "description": "Returns the start date of the currently selected range as JavaScript Date instance."
  },
  { "name": "closePicker", "visibility": "public", "description": "Closes the picker." },
  {
    "name": "formatValue",
    "visibility": "public",
    "returnValue": { "type": "string", "description": "The date as string" },
    "parameters": [
      {
        "name": "date",
        "type": "object",
        "optional": false,
        "description": "A Java Script date object to be formatted as string"
      }
    ],
    "description": "Formats a Java Script date object into a string representing a locale date according to the <code>formatPattern</code> property of the DatePicker instance"
  },
  {
    "name": "isInValidRange",
    "visibility": "public",
    "returnValue": { "type": "boolean" },
    "parameters": [
      {
        "name": "value",
        "type": "string",
        "optional": true,
        "defaultValue": "\"\"",
        "description": "A value to be checked"
      }
    ],
    "description": "Checks if a date is between the minimum and maximum date."
  },
  {
    "name": "isOpen",
    "visibility": "public",
    "returnValue": { "type": "boolean", "description": "true if the picker is open, false otherwise" },
    "description": "Checks if the picker is open."
  },
  {
    "name": "isValid",
    "visibility": "public",
    "returnValue": { "type": "boolean" },
    "parameters": [
      {
        "name": "value",
        "type": "string",
        "optional": true,
        "defaultValue": "\"\"",
        "description": "A value to be tested against the current date format"
      }
    ],
    "description": "Checks if a value is valid against the current date format of the DatePicker."
  },
  {
    "name": "openPicker",
    "visibility": "public",
    "returnValue": { "type": "Promise", "description": "Resolves when the picker is open" },
    "description": "Opens the picker."
  }
]
