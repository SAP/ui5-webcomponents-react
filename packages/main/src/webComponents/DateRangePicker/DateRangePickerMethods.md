## Attributes & Methods

This component exposes public attributes and methods. You can use them directly on the instance of the component, e.g. by using React Refs.

| Name                          | Parameters                                                                                                                                                         | Description                                                                                                                                                |
| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **dateValue (readonly)**      | &mdash;                                                                                                                                                            | <b>Note:</b> The getter method is inherited and not supported. If called it will return an empty value.                                                    |
| **dateValueUTC (readonly)**   | &mdash;                                                                                                                                                            | <b>Note:</b> The getter method is inherited and not supported. If called it will return an empty value.                                                    |
| **endDateValue (readonly)**   | &mdash;                                                                                                                                                            | Returns the end date of the currently selected range as JavaScript Date instance.                                                                          |
| **startDateValue (readonly)** | &mdash;                                                                                                                                                            | Returns the start date of the currently selected range as JavaScript Date instance.                                                                        |
| **closePicker**               | &mdash;                                                                                                                                                            | Closes the picker.                                                                                                                                         |
| **formatValue**               | <dl><dt className="methodText">**date**</dt><dd className="methodText">A Java Script date object to be formatted as string</dd><dd><code>object</code></dd></dl>   | Formats a Java Script date object into a string representing a locale date according to the <code>formatPattern</code> property of the DatePicker instance |
| **isInValidRange**            | <dl><dt className="methodText">**value**</dt><dd className="methodText">A value to be checked</dd><dd><code>string</code></dd></dl>                                | Checks if a date is between the minimum and maximum date.                                                                                                  |
| **isOpen**                    | &mdash;                                                                                                                                                            | Checks if the picker is open.                                                                                                                              |
| **isValid**                   | <dl><dt className="methodText">**value**</dt><dd className="methodText">A value to be tested against the current date format</dd><dd><code>string</code></dd></dl> | Checks if a value is valid against the current date format of the DatePicker.                                                                              |
| **openPicker**                | &mdash;                                                                                                                                                            | Opens the picker.                                                                                                                                          |
