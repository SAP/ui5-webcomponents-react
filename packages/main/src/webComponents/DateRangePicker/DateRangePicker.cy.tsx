import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { DateRangePicker } from './index';

describe('DateRangePicker', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<DateRangePicker />);
    cy.get('[ui5-daterange-picker]').should('exist');
  });

  cypressPassThroughTestsFactory(DateRangePicker);
});
