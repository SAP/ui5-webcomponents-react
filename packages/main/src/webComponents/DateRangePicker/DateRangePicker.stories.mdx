import { Canvas, Meta, Story } from '@storybook/addon-docs';
import { ArgsTableWithNote } from '@docs/ArgsTableWithNote';
import { DateRangePicker } from './index';
import { CalendarType } from '../../enums/CalendarType';
import '@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Buddhist.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Islamic.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Japanese.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Persian.js';
import { ValueState } from '../../enums/ValueState';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';
import Description from './DateRangePickerDescription.md';
import DateRangePickerMethods from './DateRangePickerMethods.md';

<Meta
  title="Inputs / DateRangePicker"
  component={DateRangePicker}
  argTypes={{
    valueStateMessage: { control: { disable: true } }
  }}
  args={{
    primaryCalendarType: CalendarType.Gregorian,
    valueState: ValueState.None
  }}
/>

<DocsHeader since="0.10.0" />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <DateRangePicker {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTableWithNote story="." />

<Description />

<DateRangePickerMethods />

<Footer />
