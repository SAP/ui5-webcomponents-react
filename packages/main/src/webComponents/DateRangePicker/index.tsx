import { CalendarType } from '@ui5/webcomponents-react/dist/CalendarType';
import { ValueState } from '@ui5/webcomponents-react/dist/ValueState';
import { withWebComponent } from '@ui5/webcomponents-react/dist/withWebComponent';
import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';
import { Ui5CustomEvent } from '@ui5/webcomponents-react/interfaces/Ui5CustomEvent';
import { Ui5DateRangePickerDomRef } from '@ui5/webcomponents-react/interfaces/Ui5DateRangePickerDomRef';
import { ReactNode } from 'react';

import '@ui5/webcomponents/dist/DateRangePicker.js';

export interface DateRangePickerPropTypes extends Omit<CommonProps, 'onChange' | 'onInput'> {
  /**
   * Determines the symbol which separates the dates. If not supplied, the default time interval delimiter for the current locale will be used.
   */
  delimiter?: string;
  /**
   * Defines the aria-label attribute for the component.
   */
  accessibleName?: string;
  /**
   * Receives id(or many ids) of the elements that label the component.
   */
  accessibleNameRef?: string;
  /**
   * Determines whether the component is displayed as disabled.
   */
  disabled?: boolean;
  /**
   * Defines the visibility of the week numbers column.
   *
   * **Note: **For calendars other than Gregorian, the week numbers are not displayed regardless of what is set.****
   */
  hideWeekNumbers?: boolean;
  /**
   * Determines the name with which the component will be submitted in an HTML form.
   *
   * **Important:** For the `name` property to have effect, you must add the following import to your project: `import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";`
   *
   * **Note:** When set, a native `input` HTML element will be created inside the component so that it can be submitted as part of an HTML form. Do not use this property unless you need to submit a form.
   */
  name?: string;
  /**
   * Defines a short hint, intended to aid the user with data entry when the component has no value.
   *
   * **Note:** When no placeholder is set, the format pattern is displayed as a placeholder. Passing an empty string as the value of this property will make the component appear empty - without placeholder or format pattern.
   */
  placeholder?: string;
  /**
   * Determines whether the component is displayed as read-only.
   */
  readonly?: boolean;
  /**
   * Defines whether the component is required.
   */
  required?: boolean;
  /**
   * Defines a formatted date value.
   */
  value?: string;
  /**
   * Defines the value state of the component.
   *
   * Available options are:
   *
   * *   `None`
   * *   `Error`
   * *   `Warning`
   * *   `Success`
   * *   `Information`
   */
  valueState?: ValueState | keyof typeof ValueState;
  /**
   * Determines the format, displayed in the input field.
   */
  formatPattern?: string;
  /**
   * Determines the maximum date available for selection.
   */
  maxDate?: string;
  /**
   * Determines the Ð¼inimum date available for selection.
   */
  minDate?: string;
  /**
   * Sets a calendar type used for display. If not set, the calendar type of the global configuration is used.<br/>__Note:__ Calendar types other than Gregorian must be imported manually:<br />`import "@ui5/webcomponents-localization/dist/features/calendar/{primaryCalendarType}.js";`
   */
  primaryCalendarType?: CalendarType | keyof typeof CalendarType;
  /**
   * Defines the value state message that will be displayed as pop up under the component.
   *
   * **Note:** If not specified, a default text (in the respective language) will be displayed.
   * **Note:** The `valueStateMessage` would be displayed, when the component is in `Information`, `Warning` or `Error` value state.
   *
   * __Note:__ When passing a custom React component to this prop, you have to make sure your component reads the `slot` prop and appends it to the most outer element of your component.
   * Learn more about it [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base--page#adding-custom-components-to-slots).
   */
  valueStateMessage?: ReactNode;
  /**
   * Fired when the input operation has finished by pressing Enter or on focusout.
   */
  onChange?: (event: Ui5CustomEvent<HTMLInputElement>) => void;
  /**
   * Fired when the value of the component is changed at each key stroke.
   */
  onInput?: (event: Ui5CustomEvent<HTMLInputElement>) => void;
}

/**
 * The DateRangePicker enables the users to enter a localized date range using touch, mouse, keyboard input, or by selecting a date range in the calendar.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/DateRangePicker" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const DateRangePicker = withWebComponent<DateRangePickerPropTypes, Ui5DateRangePickerDomRef>(
  'ui5-daterange-picker',
  [
    'delimiter',
    'accessibleName',
    'accessibleNameRef',
    'name',
    'placeholder',
    'value',
    'valueState',
    'formatPattern',
    'maxDate',
    'minDate',
    'primaryCalendarType'
  ],
  ['disabled', 'hideWeekNumbers', 'readonly', 'required'],
  ['valueStateMessage'],
  ['change', 'input']
);

DateRangePicker.displayName = 'DateRangePicker';

DateRangePicker.defaultProps = {
  disabled: false,
  hideWeekNumbers: false,
  placeholder: undefined,
  readonly: false,
  required: false,
  valueState: ValueState.None
};

export { DateRangePicker };
