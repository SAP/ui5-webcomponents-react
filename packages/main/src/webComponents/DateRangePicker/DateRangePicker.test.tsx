import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { DateRangePicker } from '@ui5/webcomponents-react/dist/DateRangePicker';
import React from 'react';

describe('DateRangePicker', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<DateRangePicker />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(DateRangePicker);
});
