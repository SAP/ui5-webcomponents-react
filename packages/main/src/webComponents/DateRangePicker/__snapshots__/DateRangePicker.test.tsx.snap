// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-daterange-picker
    ui5-daterange-picker=""
    value-state="None"
  />
</DocumentFragment>
`;
